app/scripts/lib/rpc-method-middleware/handlers/send-metadata.ts

```diff
@@ -0,0 +1,84 @@
+import { ethErrors } from 'eth-rpc-errors';
+import type { SubjectType } from '@metamask/permission-controller';
+import type {
+  Json,
+  JsonRpcParams,
+  JsonRpcRequest,
+  JsonRpcResponse,
+} from '@metamask/utils';
+import type {
+  JsonRpcEngineEndCallback,
+  JsonRpcEngineNextCallback,
+} from '@metamask/json-rpc-engine';
+import { MESSAGE_TYPE } from '../../../../../shared/constants/app';
+import { AddSubjectMetadata, HandlerWrapperType } from './handlers-helper';
+
+type SendMetadataOptionsType = {
+  addSubjectMetadata: AddSubjectMetadata;
+  subjectType: SubjectType;
+};
+
+type SendMetadataType<
```

Comment:
nit: What do you think about renaming the type name here. Appending "type" doesn't add any useful information that isn't provided by the CamelCase styling.

```suggestion
type SendMetadata<
```

```suggestion
type SendMetadataConstraint<
```

Reply:
updated. :)